sympy/physics/vector/frame.py

```diff
@@ -554,45 +546,145 @@ def orient(self, parent, rot_type, amounts, rot_order=''):
         The following two lines show how the sense of the rotation can be
         defined. Both lines produce the same result.
 
-        >>> B.orient(N, 'Axis', (q1, -N.x))
-        >>> B.orient(N, 'Axis', (-q1, N.x))
+        >>> B.orient_axis(N, (q1, -N.x))
+        >>> B.orient_axis(N, (-q1, N.x))
+
+        """
+
+        from sympy.physics.vector.functions import dynamicsymbols
+        _check_frame(parent)
+
+        amounts = list(amounts)
+        for i, v in enumerate(amounts):
+            if not isinstance(v, Vector):
+                amounts[i] = sympify(v)
+        if not (isinstance(amounts, (list, tuple)) & (len(amounts) == 2)):
+            raise TypeError('Amounts are a list or tuple of length 2')
+        theta = amounts[0]
+        axis = amounts[1]
+        axis = _check_vector(axis)
+        parent_orient_axis = []
+
+        if not axis.dt(parent) == 0:
+            raise ValueError('Axis cannot be time-varying')
+        axis = axis.express(parent).normalize()
+        axis = axis.args[0][0]
+        parent_orient_axis = ((eye(3) - axis * axis.T) * cos(theta) +
+                            Matrix([[0, -axis[2], axis[1]],
+                                    [axis[2], 0, -axis[0]],
+                                    [-axis[1], axis[0], 0]]) *
+                            sin(theta) + axis * axis.T)
+
+        self._dcm(parent, parent_orient_axis)
+
+        thetad = (amounts[0]).diff(dynamicsymbols._t)
+        wvec = thetad * amounts[1].express(parent).normalize()
+        self._ang_vel_dict.update({parent: wvec})
+        parent._ang_vel_dict.update({self: -wvec})
+        self._var_dict = {}
+
+    def orient_dcm(self, parent, amounts):
```

Comment:
This could likely have a better function name. Some ideas:

- orient_explicit (because we are explicit defining the direction cosine matrix (rotation matrix))
- orient_direction_cosine_matrix
- orient_from_direction_cosine_matrix
- orient_rotation_matrix
- orient_from_rotation_matrix